Sources/TuistCore/ValueGraph/ValueGraph.swift

```diff
@@ -16,15 +22,114 @@ public struct ValueGraph: Equatable {
     public let targets: [AbsolutePath: [String: Target]]
 
     /// A dictionary that contains the one-to-many dependencies that represent the graph.
-    public let dependencies: [ValueGraphDependency: [ValueGraphDependency]]
+    public let dependencies: [ValueGraphDependency: Set<ValueGraphDependency>]
 
-    public init(projects: [AbsolutePath: Project],
+    public init(name: String,
+                path: AbsolutePath,
+                projects: [AbsolutePath: Project],
                 packages: [AbsolutePath: [String: Package]],
                 targets: [AbsolutePath: [String: Target]],
-                dependencies: [ValueGraphDependency: [ValueGraphDependency]]) {
+                dependencies: [ValueGraphDependency: Set<ValueGraphDependency>]) {
+        self.name = name
+        self.path = path
         self.projects = projects
         self.packages = packages
         self.targets = targets
         self.dependencies = dependencies
     }
+
+    init(name: String, path: AbsolutePath, cache: GraphLoaderCaching) {
```

Comment:
is there a need to also have a cache initialiser? May help remove some duplicate code if we limit it to `Graph` seeing we can make a graph from the cache.

Reply:
You are right, we can keep only the graph initializer.